rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -397,6 +404,7 @@ NodeParameters::declare_parameter(
     on_parameters_set_callback_,
     &parameter_event,
     ignore_override);
+  parameter_modification_enabled_ = true;
```

Comment:
I originally did this because I wanted to set `parameter_modification_enabled_` to true as soon as possible (and not wait for the end of the function).  But looking at it again, I think a better way is to just introduce a local scope and let the `RCLCPP_SCOPE_EXIT` take care of it.  I'll make that change.